File: 370__Range_Addition.py
class Solution(object):
    def getModifiedArray(self, length, updates):
        """
        :type length: int
        :type updates: List[List[int]]
        :rtype: List[int]
        """
        result = [0] * length
        for start, end, increment in updates:
            result[start] += increment
            if end + 1 < length:
                result[end + 1] -= increment

        current = 0
        for i, val in enumerate(result):
            current += val
            result[i] = current
        return result

File: Java/370__Range_Addition.java
class Solution {
    public int[] getModifiedArray(int length, int[][] updates) {
        int[] result = new int[length];
        // Iterate updates and use answer array to store two-end info.
        for (int[] update : updates) {
            int l_pos = update[0], r_pos = update[1];
            result[l_pos] += update[2];
            // Keep update in the range by removing the increment after.
            if (r_pos < length - 1) {
                // Ignore the rightmost update.
                result[r_pos + 1] -= update[2];
            }
        }
        int sum = 0;
        for (int i = 0; i < length; i++) {
            sum += result[i];
            result[i] = sum;
        }
        return result;
    }
}
